### DISCUSSION TITLE
Switch Relay connection lost while updating to 12.x

### DISCUSSION THREAD

**Discussion by Taxxian** [arendst/Tasmota#17668] [Q&A]
Hi,

I have an old Carport door and wanted to bring that into my Smarthome.

So I bought a Wemos D1 board:
GPIO 3 and 4 are set to Relay1/2 and they switch the opening circuit und the emergency lock, thats works fine still.
GPIO 2 is set to Switch3 (yes not the ideal pin) and connected to a relay which is connected to a AC/DC rectifier which is connected to the 24V AC Light port, so I can detect if the thing is open or closed.
Then there is GPIO 0 which is connected to nothing but set to Relay3, so I can see the status of Switch3 in MQQT.

That worked fine for many years until a few weeks ago when I updated from Tasmota 9.x to 12.x.
Since this update I can still see messages in the console when opening the door, but the switch3 sensor in Home Assistant shows "unavailable" and Relay3 no longer reacts to changes on Switch3.

What can I to to connect the two again?

Regards

Martin

**Comment by barbudor** [comment]
You are probably facing the obsolescence of old HA integration (Setoption19 1)
This is now out and replaced in favor of Tasmota-Discovery which requires HaTasmota plug in HA

A switch of gpio 2 is really really a bad idea as it could prevent the esp to boot when driven to ground

**Reply by Taxxian** [reply]
I just have to open the door when installing updates, its fine *lol*

Thanks, I will check the new integration, but I am quite sure the "Toggle 3" in the Tasmota web interface was reacting to opening and closing the door until I updated to 12.2. And that has nothing to do with HA.

**Reply by Taxxian** [reply]
I just checked the Tasmota Integration:

The "Switch 3" sensor is entirely missing there and "Relay 3" does not react to opening and closing, so it does not help...

**Reply by barbudor** [reply]
Can you share your detailled configuration ?
- output of command: `Backlog Template; Module; GPIO 255`
- Switchmode for each switches
- Any rule ? output of `Rule0`

Note that there shouldn't be any "hole" in your numbering of Relay, Buttons or Switches
Normally you not supposed to have a Switch_3 if you haven't a Switch_1 and a Switch_2

**Reply by Taxxian** [reply]
Of course, I tried chaning my switch3 to switch1 but that did not change anything, the corresponding virtual relay does not react.

00:25:27.048 MQT: Carport/stat/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,4704],"FLAG":0,"BASE":18}
00:25:27.259 MQT: Carport/stat/RESULT = {"Module":{"18":"Generic"}}
00:25:27.513 MQT: Carport/stat/RESULT = {"GPIO0":{"226":"Relay3"},"GPIO1":{"0":"None"},"GPIO2":{"162":"Switch3"},"GPIO3":{"0":"None"},"GPIO4":{"225":"Relay2"},"GPIO5":{"224":"Relay1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

00:26:30.845 MQT: Carport/stat/RESULT = {"SwitchMode1":3}

00:26:55.462 MQT: Carport/stat/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
00:26:55.466 MQT: Carport/stat/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
00:26:55.469 MQT: Carport/stat/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

Just writing a rule that connects switch3 to relay3 is probably the best solution, but why did it work for years (since Tasmota 5.x) without rules?

When I open/close it reads:
00:30:11.069 MQT: alarm/cmnd/POWER3 = ON
00:30:12.074 MQT: alarm/cmnd/POWER3 = OFF
00:30:13.104 MQT: alarm/cmnd/POWER3 = ON
00:30:14.160 MQT: alarm/cmnd/POWER3 = OFF
00:30:15.212 MQT: alarm/cmnd/POWER3 = ON
00:30:16.275 MQT: alarm/cmnd/POWER3 = OFF
So the switch works, while the relay is always off.
